cmd/state-svc/test/integration/svc_int_test.go

```diff
@@ -0,0 +1,153 @@
+	"github.com/ActiveState/cli/internal/svcctl"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/shirou/gopsutil/process"
+	"github.com/stretchr/testify/suite"
+)
+
+type SvcIntegrationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *SvcIntegrationTestSuite) TestStartStop() {
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cp := ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Service cannot be reached")
+	cp.ExpectExitCode(1)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("start"))
+	cp.Expect("Starting")
+	cp.ExpectExitCode(0)
+	time.Sleep(500 * time.Millisecond) // wait for service to start up
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Checking")
+
+	// Verify it created a socket file.
+	sockFile := svcctl.NewIPCSockPathFromGlobals().String()
+	suite.True(fileutils.TargetExists(sockFile), "socket file '"+sockFile+"' does not exist")
+
+	// Verify the server is running on its reported port.
+	cp.ExpectRe("Port:\\s+:\\d+\\s")
+	portRe := regexp.MustCompile("Port:\\s+:(\\d+)")
+	port := portRe.FindStringSubmatch(cp.TrimmedSnapshot())[1]
+	_, err := net.Listen("tcp", "localhost:"+port)
+	suite.Error(err)
+
+	// Verify it created and wrote to its reported log file.
+	cp.ExpectRe("Log:\\s+.+?\\.log")
+	logRe := regexp.MustCompile("Log:\\s+(.+?\\.log)")
+	logFile := logRe.FindStringSubmatch(cp.TrimmedSnapshot())[1]
+	suite.True(fileutils.FileExists(logFile), "log file '"+logFile+"' does not exist")
+	suite.True(len(fileutils.ReadFileUnsafe(logFile)) > 0, "log file is empty")
+
+	cp.ExpectExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("stop"))
+	cp.Expect("Stopping")
+	cp.ExpectExitCode(0)
+	time.Sleep(500 * time.Millisecond) // wait for service to stop
+
+	// Verify it deleted its socket file and the port is free.
+	suite.False(fileutils.TargetExists(sockFile), "socket file was not deleted")
+	server, err := net.Listen("tcp", "localhost:"+port)
+	suite.NoError(err)
+	server.Close()
+}
+
+func (suite *SvcIntegrationTestSuite) TestSignals() {
+	if runtime.GOOS == "windows" {
+		suite.T().Skip("Windows does not support signal sending.")
+	}
+
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	// SIGINT (^C)
+	cp := ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("foreground"))
+	cp.Expect("Starting")
+	time.Sleep(1 * time.Second) // wait for the service to start up
+	cp.Signal(syscall.SIGINT)
+	cp.Expect("caught a signal: interrupt")
+	cp.ExpectNotExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Service cannot be reached")
+	cp.ExpectExitCode(1)
+
+	sockFile := svcctl.NewIPCSockPathFromGlobals().String()
+	if runtime.GOOS != "linux" {
+		// TODO: this fails on only Linux for some reason, but it is not reproducable outside of CI
```

Comment:
Please file a story, or convey it in an existing story, then leave a link in the comment instead of a TODO.